File: codeforces/virtual/250/2.java
/* Author: Hemanth Kumar Veeranki */
/* Handle: harry7 */
import java.io.*;
import java.util.*;
import java.math.*;
class Main{
	public static void main(String[] args){
		Scanner sc=new Scanner(System.in); /*sc for taking input*/
		PrintStream op=System.out;/*op for printing output*/
		int a,b,l;
		a=sc.nextInt();
		b=sc.nextInt();
		Vector<Integer> ans=new Vector<>();
		while(b>0){
			l=b&(-b);
			if(a>=l){
				a-=l;
				ans.add(b);
			}
			b--;
		}
		if(a!=0)op.println("-1");
		else{
			op.println(ans.size());
			Iterator i = ans.iterator();
			while (i.hasNext()) op.print(i.next()+" ");
		}
	}
}

File: codeforces/virtual/295/2.py
""" 
    Author: Hemanth Kumar Veeranki
    Handle:harry7

"""
#!/usr/bin/python
n,m = map(int,raw_input().split())
if n >= m:
    print n-m
else:
    used = [False]*50001
    used[n] = True
    que = [[n,0]]
    while 1:
        k,i = que.pop(0)
        if k == m:
            break
        if k < m and not used[2*k]:
            used[2*k] = True
            que.append([2*k,i+1])
        if 1 < k and not used[k-1]:
            used[k-1] = True
            que.append([k-1,i+1])
    print i
